benchmark/BenchmarkConfiguration.h

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (March of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include "util/json.h"
+#include <type_traits>
+
+/*
+ * A rather basic wrapper for nlohmann::json, which only allows reading of
+ * information and setting the configuration by parsing strings.
+ */
+class BenchmarkConfiguration{
+ // No real reason, to really build everything ourselves, when the
+ // nlohmann::json object already containes everything, that we could need.
+ nlohmann::json data_;
+
+public:
+
+ /*
+  * @brief Returns, if an configuration option was set.
+  *
+  * @tparam Index,Indexes The types of the keys, that you gave. Deduction
+  *  should be able to handle it.
+  * 
+  * @param index,indexes The keys for looking up the configuration option.
+  *  Look at the documentation of `nlohmann::basic_json::at`, if you want
+  *  to see, what's possible.
+  */
+ template<typename Index, typename... Indexes>
+ bool isOptionSet(const Index& index, const Indexes&... indexes) const{
+  // To reduce duplication. Is a key held by a given json object?
+  auto isKeyValid = []<typename Key>(const auto& jsonObject, const Key& key){
+   // If `Key` is a number, it can only be a valid key for the `jsonObject`
+   // , if the `jsonObject` is an array.
+   if constexpr (std::is_integral<Key>::value){
+    return jsonObject.is_array() && (key < jsonObject.size());
+   }else{
+    return jsonObject.contains(key);
```

Comment:
I'm well aware, that it's `if the object is an array, then the key must be integral.`, but the `jsonObject` isn't `constexpr`, because it's an argument. So, I can't ask that at runtime.

Secondly, what does a hash map have to do with anything? As far as I can tell, in `nlohmann::json` the keys for array elements and object element are disjunctive: https://json.nlohmann.me/api/basic_json/at/#version-history
Honestly, I'm not sure, the documentation is horrible, when it comes to finding any information's about types.